phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/IndexScrutinyMapper.java

```diff
@@ -254,6 +291,47 @@ private void processBatch(Context context)
         }
     }
 
+    private void checkIfInvalidRowsExpired(Context context,
+            Map<String, Pair<Long,
+            List<Object>>> targetPkToSourceValues) throws SQLException, IOException {
+        long ttl = getTableTtl();
+        Set<Map.Entry<String, Pair<Long,
+                                List<Object>>>> entrySet = targetPkToSourceValues.entrySet();
+
+        Iterator<Map.Entry<String, Pair<Long,
+                        List<Object>>>> itr = entrySet.iterator();
+
+        // iterate and remove items simultaneously
+        while(itr.hasNext()) {
+            Map.Entry<String, Pair<Long, List<Object>>> entry = itr.next();
+            Pair<Long, List<Object>> sourceValues = entry.getValue();
+            Long sourceTS = sourceValues.getFirst();
+            if (hasRowExpiredOnSource(sourceTS, ttl)) {
+                context.getCounter(PhoenixScrutinyJobCounters.EXPIRED_ROW_COUNT).increment(1);
+                itr.remove();
+            }
+        }
+    }
+
+    private boolean hasRowExpiredOnSource(Long sourceTS, Long ttl) {
+        long currentTS = EnvironmentEdgeManager.currentTimeMillis();
+        if (sourceTS + ttl < currentTS) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    private long getTableTtl() throws SQLException, IOException {
+        PTable psourceTable = PhoenixRuntime.getTable(connection, qSourceTable);
+        Admin admin = connection.unwrap(PhoenixConnection.class).getQueryServices().getAdmin();
+        String fullTableName = SchemaUtil.getQualifiedTableName(psourceTable.getSchemaName().toString(),
+                psourceTable.getTableName().toString());
+        HTableDescriptor tableDesc = admin.getTableDescriptor(TableName.valueOf(fullTableName));
+        long ttl = tableDesc.getFamily(SchemaUtil.getEmptyColumnFamily(psourceTable)).getTimeToLive();
```

Comment:
What is the TTL for tables that don't have it set explicitly? If it's LONG_MAX, then please make sure hasRowExpiredOnSource() does not overflow when doing sourceTS + ttl.